valohai_yaml/objs/pipelines/pipeline.py

```diff
@@ -45,6 +45,16 @@ def get_data(self) -> SerializedDict:
         return data
 
     def lint(self, lint_result: LintResult, context: LintContext) -> None:
+        # ensure unique names for nodes
+        node_name_counts = {}
+        for node in self.nodes:
+            node_name_counts.setdefault(node.name, 0)
+            node_name_counts[node.name] += 1
```

Comment:
Import `itertools.Counter` and
```suggestion
        # ensure unique names for nodes
        node_name_counts = Counter(node.name for node in self.nodes)
```